Fixed Issues:

saurabh-pingale-cognito opened the pull request:
- Added country configuration to show products based on country
- Added export to excel sheet in shipment export
- Displayed orders in admin
- Added export to excel sheet in orders
- Added order filtering
- Added updated order status in admin
- Optimized queries
- Resolved comments from PR
- Fixed playwright test

saurabh78crypto committed:
Fixed Issues:
- Added country configuration to show products based on country
- Added export to excel sheet in shipment export
- Displayed orders in admin
- Added export to excel sheet in orders
- Added order filtering
- Added updatred order status in admin

BalanaguYashwanth commented on backend/warehouse-server/src/ecommerce/entities/ecommerce-order.entity.ts:
```diff
@@ -71,6 +71,15 @@ export class EcommerceOrder extends BaseTimestampEntity {
   total_amount: number;
 
+  @Column({ type: 'varchar', nullable: true })
+  payment_gateway: string;
```
``payment_gateway`` can't be nullable true, its mandatory field

BalanaguYashwanth commented on backend/warehouse-server/src/ecommerce/entities/ecommerce-order.entity.ts:
```diff
@@ -71,6 +71,15 @@ export class EcommerceOrder extends BaseTimestampEntity {
+  payment_gateway: string;
+
+  @Column({ type: 'varchar', nullable: true })
+  payment_mode: string;
```
``payment_mode`` mandatory field

BalanaguYashwanth commented on backend/warehouse-server/src/ecommerce/services/ecommerce-order.service.ts:
```diff
@@ -22,14 +24,19 @@ interface CurrencyInfo {
 }
 
-interface CashfreePaymentUpdate {
+interface CashfreePayment {
```
``CashfreePayment`` we will be might moving razourpay, so hardcoding ``razourpay``or ``cashfree``won't make sense, 

so please add correct naming conventions

BalanaguYashwanth commented on backend/warehouse-server/src/ecommerce/services/ecommerce-order.service.ts:
```diff
@@ -22,14 +24,19 @@ interface CurrencyInfo {
 
-interface CashfreePaymentUpdate {
+interface CashfreePayment {
   cf_payment_id: string;
```
``cf_payment_id`` just add payment_id, if we replace with ``razourpay`` then it has ``rz_payment_id``, 

so again we can't modify, please check according do naming convention changes

saurabh-pingale-cognito replied:
Here, we can't use the `payment_id` directly because the `successfulPayment` object returns the `cf_payment_id`

`
{
  auth_id: null,
  authorization: null,
  bank_reference: '1234567890',
  cf_payment_id: '5114923470933',
  entity: 'payment',
  error_details: null,
  international_payment: { international: false },
  is_captured: true,
  order_amount: 94.82,
  order_currency: 'INR',
  order_id: 'ORD-1765524918092-Q4I97I0',
  payment_amount: 94.82,
  payment_completion_time: '2025-12-12T13:06:00+05:30',
  payment_currency: 'INR',
  payment_gateway_details: {
    gateway_name: 'CASHFREE',
    gateway_order_id: '2201939344',
    gateway_payment_id: '5114923470933',
    gateway_order_reference_id: null,
    gateway_status_code: null,
    gateway_settlement: 'cashfree',
    gateway_reference_name: null
  },
  payment_group: 'upi',
  payment_message: 'Simulated response message',
  payment_method: {
    upi: {
      channel: 'collect',
      upi_id: 'testsuccess@gocash',
      upi_instrument: 'UPI',
      upi_instrument_number: '',
      upi_payer_account_number: '',
      upi_payer_ifsc: ''
    }
  },
  payment_offers: null,
  payment_status: 'SUCCESS',
  payment_surcharge: {
    payment_surcharge_service_charge: 0,
    payment_surcharge_service_tax: 0
  },
  payment_time: '2025-12-12T13:05:42+05:30'
}`

BalanaguYashwanth commented on backend/warehouse-server/src/ecommerce/services/ecommerce-order.service.ts:
```diff
@@ -41,19 +48,27 @@ export class OrderService {
-    const mode = process.env.CASHFREE_MODE;
+    this.appId = process.env.CASHFREE_APP_ID!;
+    this.secretKey = process.env.CASHFREE_SECRET_KEY!;
+    this.mode = process.env.CASHFREE_MODE!;
```
```
this.appId = process.env.CASHFREE_APP_ID!;
this.secretKey = process.env.CASHFREE_SECRET_KEY!;
this.mode = process.env.CASHFREE_MODE!;
```
Add proper naming convention

BalanaguYashwanth commented on backend/warehouse-server/src/ecommerce/services/ecommerce-order.service.ts:
```diff
@@ -41,19 +48,27 @@ export class OrderService {
     }
 
+    this.baseUrl =
+      this.mode === 'production'
```
``'production'`` or ``sandbox`` should come from .env

saurabh-pingale-cognito replied:
Here, mode coming from the .env only
And based on that we're conditionally using the base url and CFenvironment

`
this.gatewayMode = process.env.CASHFREE_MODE!;

 this.gatewayBaseUrl =
      this.gatewayMode === 'production'
        ? 'https://api.cashfree.com/pg'
        : 'https://sandbox.cashfree.com/pg';

    this.cashfree = new Cashfree(
      this.gatewayMode === 'production'
        ? CFEnvironment.PRODUCTION
        : CFEnvironment.SANDBOX,
      this.gatewayAppId,
      this.gatewaySecretKey,
    );
`

BalanaguYashwanth commented on backend/warehouse-server/src/ecommerce/services/ecommerce-order.service.ts:
```diff
@@ -175,6 +190,7 @@ export class OrderService {
       shipping_amount: roundedShipping,
       tax_amount: roundedTax,
       total_amount: roundedTotal,
+      payment_gateway: 'cashfree',
```
payment_gateway of `` 'cashfree',`` add in constants and use these names

BalanaguYashwanth commented on backend/warehouse-server/src/ecommerce/services/ecommerce-order.service.ts:
```diff
@@ -302,6 +318,26 @@ export class OrderService {
+    const orders = await this.orderRepository.find({
+      relations: ['items', 'user'],
+      order: { created_at: 'DESC' },
+    });
```
```
  const orders = await this.orderRepository.find({
      relations: ['items', 'user'],
      order: { created_at: 'DESC' },
    });
```
Do query optiimization

BalanaguYashwanth commented on backend/warehouse-server/src/ecommerce/services/ecommerce-order.service.ts:
```diff
@@ -302,6 +318,26 @@ export class OrderService {
+      order: { created_at: 'DESC' },
+    });
+
+    return orders.map((order) => ({
```
``Avoid looping``

BalanaguYashwanth commented on backend/warehouse-server/src/ecommerce/services/ecommerce-order.service.ts:
```diff
@@ -380,23 +416,62 @@ export class OrderService {
     const order = await this.findOne(id);
     order.status = status;
+
+    if (comment !== undefined) {
```
if ``comment`` is null or None, then this will fail, 

so ``if(comment)`` then add block

BalanaguYashwanth commented on frontend/admin/src/pages/Orders.tsx:
```diff
@@ -2,156 +2,142 @@ import type React from "react";
-    { value: 'Shipped', label: 'Shipped' },
-    { value: 'Delivered', label: 'Delivered' },
-    { value: 'Cancelled', label: 'Cancelled' },
+  const statusOptions: StatusOption[] = [
```
Move this to constants ``statusOptions``

BalanaguYashwanth commented on frontend/warehouse-app/src/app/ecommerce/page.tsx:
```diff
@@ -122,11 +126,11 @@ export default function Ecommerce() {
     setError(null);
     hasFetched.current = false;
-    initializeEcommerceData(locationData.currencyInfo.code);
+    initializeEcommerceData(locationData.currencyInfo.code, locationData.location.countryCode);
```
safe handle 
``locationData.location.countryCode``
&
``locationData.currencyInfo.code``

saurabh78crypto committed:
Refator code:
- optimized queries
- resolved comments from PR
- fixed playwright

saurabh78crypto committed:
Refactor code:
- Fixed sync to cart
- Fixed cart item selection
- Added sucess popup after successful payment
- Optimized queries
- Removed unnecessary mapping

saurabh78crypto committed:
Fixed issues:
- In all category rendered all products without mentioning the category
- Fixed category selection
- Hide the scroll button if there are no more categories outside the client viewport
- Added palakart image in the client login page
- Fixed user preference service to restrict one preference per user
- Removed unnecessary fields from the order entity (subtotal, discount_percentage, shipping_amount, tax_amount) and converted the total_amount to USD before storing it in the DB
- Fixed loaders in the EditProfileModal
- Fixed spacing of columns in admin orders page
- Fixed export shipment export without selecting any box (will export data from all the boxes)
- Fixed cart item selection
- Updated the UI of order item in the client orders page

BalanaguYashwanth committed:
Merge pull request #71 from Cognito-Innovations/refator/code

Refactor code:
